src/ingest-pipeline/md/data_collection_types/multiassay_metadatatsv_data_collection.py

```diff
@@ -0,0 +1,143 @@
+        containing data of this collection type?
+        """
+        offsetdir = cls.find_top(path, cls.top_target, cls.dir_regex)
+        print('Checking for multiple metadata.tsv at top level')
+        if offsetdir is None:
+            return False
+        candidates = os.listdir(os.path.join(path, offsetdir))
+        return (len(candidates) > 1 and candidates[0].endswith('-metadata.tsv'))
+
+    def __init__(self, path):
+        """
+        path is the top level directory of the collection
+        """
+        super().__init__(path)
+        self.offsetdir = self.find_top(self.topdir, self.top_target, self.dir_regex)
+        assert self.offsetdir is not None, 'Wrong dataset type?'
+
+    def collect_metadata(self):
+        auth_tok = os.getenv('AUTH_TOK')
+        ingest_api_url = os.getenv('INGEST_API_URL')
+        md_type_tbl = self.get_md_type_tbl()
+        rslt = {}
+        cl = []
+        for match, md_type in self.expected_files + self.optional_files:
+            print('collect match %s' % match.format(offsetdir=self.offsetdir))
+            for fpath in glob.iglob(os.path.join(self.topdir,
+                                                 match.format(offsetdir=self.offsetdir))):
+                print('collect from path %s' % fpath)
+                this_md = md_type_tbl[md_type](fpath).collect_metadata()
+                fname = os.path.basename(fpath)
+
+                if this_md is not None:
+                    # Send the metadata file through to the assay classifier to see whether it's the multi-assay metadata or not.
+                    # If it is a multi-assay, cl.extend
+                    rslt[os.path.relpath(fpath, self.topdir)] = this_md
+                    headers = {
+                        "authorization": f"Bearer {auth_tok}",
+                        'content-type': 'application/json',
+                        'X-Hubmap-Application': 'ingest-pipeline',
+                    }
+
+                    try:
+                        response = requests.post(f'{ingest_api_url}assaytype', headers=headers,
+                                                 data=json.dumps(this_md[0]))
+                        response = response.json()
+                    except HTTPError as e:
+                        if e.response.status_code == codes.unauthorized:
+                            raise RuntimeError('ingest_api_connection authorization was rejected?')
+                        else:
+                            print('benign error')
+                            return None
+
+                    if 'metadata' in fname and fname.endswith('.tsv'):
+                        assert isinstance(this_md, list), 'metadata.tsv did not produce a list'
+                        if 'must-contain' in response:
+                            print('MULTI ASSAY FOUND')
+                            for rec in this_md:
+                                for key in ['dataset_type', 'data_path', 'contributors_path']:
```

Comment:
Is this needed? this step happens after the metadata.tsvs were already validated

Reply:
Are you referring to the for loop that checks the keys? I can remove it, was just copy-pasta from the other data collection class

Reply:
actually the validation of required columns in the metadata file